kmyth_log/src/kmyth_log.c

```diff
@@ -0,0 +1,350 @@
+/**
+ * @file  kmyth_log.c
+ *
+ * @brief Implements kmyth logging library.
+ */
+
+#include "kmyth_log.h"
+
+#include <syslog.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <time.h>
+#include <stdarg.h>
+#include <string.h>
+#include <stdbool.h>
+
+static struct log_params log_settings = {
+  .app_name = DEFAULT_APP_NAME,
+  .app_name_len = strlen(DEFAULT_APP_NAME),
+  .app_version = DEFAULT_APP_VERSION,
+  .app_version_len = strlen(DEFAULT_APP_VERSION),
+  .applog_path = DEFAULT_APPLOG_PATH,
+  .applog_path_len = strlen(DEFAULT_APPLOG_PATH),
+  .applog_output_mode = APPLOG_OUTPUT_MODE_DEFAULT,
+  .applog_severity_threshold = APPLOG_SEVERITY_THRESHOLD_DEFAULT,
+  .syslog_facility = SYSLOG_FACILITY_DEFAULT,
+  .syslog_severity_threshold = SYSLOG_SEVERITY_THRESHOLD_DEFAULT,
+};
+
+//############################################################################
+// set_app_name()
+//############################################################################
+void set_app_name(char *new_app_name)
+{
+  bool truncated = false;
+
+  if (strlen(new_app_name) <= MAX_APP_NAME_LEN)
```

Comment:
This should probably use strnlen and set the maximum value to the appropriate maximum size from the header. Since I think I'm going to make this comment a few places from here on out I'll just comment "strnlen"

Reply:
Yes, I think I corrected these. I also put the value of the strnlen() call into a temp_len variable rather than call strnlen multiple times.